src/postgres/write.rs

```diff
@@ -141,10 +142,24 @@ impl DataSink for PostgresDataSink {
         while let Some(batch) = data.next().await {
             let batch = batch.map_err(check_and_mark_retriable_error)?;
 
-            if !self
-                .postgres
-                .schema
-                .equivalent_names_and_types(batch.schema_ref())
+            let postgres_fields = &self.postgres.schema.fields;
+            let batch_fields = batch.schema_ref().fields();
+
+            if !postgres_fields
+                .iter()
+                .zip(batch_fields.iter())
+                .all(|(pg_field, batch_field)| {
+                    pg_field.name() == batch_field.name()
+                        && DFSchema::datatype_is_semantically_equal(
+                            pg_field.data_type(), // for the purposes of PostgreSQL, LargeUtf8 is equivalent to Utf8
+                            // because Postgres physically cannot store anything larger than 1Gb in text (VARCHAR)
+                            if batch_field.data_type() == &DataType::LargeUtf8 {
+                                &DataType::Utf8
+                            } else {
+                                batch_field.data_type()
+                            },
+                        )
+                })
```

Comment:
This normalizes the `batch_field` type from `LargeUtf8` to `Utf8` - but what if the `pg_field` type is `LargeUtf8`? I think a better approach would be to collect all of the types from each field, and normalize them, then run through the existing `equivalent_names_and_types` check.